src/versions.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
+const path = require('path');
+const rimraf = require('rimraf');
+
+const rootDir = path.join(__dirname, '..');
+
+/**
+ * @param {string} version 
+ */
+function removeVersion(version) {
+  for (const lang of ['nodejs', 'python', 'java', 'dotnet']) {
+    rimraf.sync(path.join(rootDir, lang, 'versioned_sidebars', `version-${version}-sidebars.json`));
+    rimraf.sync(path.join(rootDir, lang, 'versioned_docs', `version-${version}`));
+    const versions = JSON.parse(fs.readFileSync(path.join(rootDir, lang, 'versions.json'), 'utf-8'));
+    const newVersions = versions.filter(v => v !== version);
+    fs.writeFileSync(path.join(rootDir, lang, 'versions.json'), JSON.stringify(newVersions, null, 2) + '\n');
+  }
+}
+
+/**
+ * @param {string} version
+ * @param {number} keepRecentMinorVersions
+ */
+function removeRecentVersions(version, keepRecentMinorVersions) {
+  const currentMinorVersion = parseInt(version.split('.')[1], 10);
```

Comment:
You assume that all our versions are `1.*`. Despite the fact that this is probably true and we never intend to ship version 2.0, this will back-fire if we do ship `2.0` some time later.

Let's guard against it?